src/date-range-picker/embedded.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef } from 'react';
+import Calendar, { DayIndex } from './calendar';
+import InternalBox from '../box/internal';
+import SpaceBetween from '../space-between/index.js';
+
+import RelativeRangePicker from './relative-range';
+import ModeSwitcher from './mode-switcher';
+import { useDateRangePicker } from './use-date-range-picker';
+import { DateRangePickerBaseProps } from './interfaces';
+import { fireNonCancelableEvent } from '../internal/events/index.js';
+import { DateRangePickerProps, Focusable } from './interfaces';
+
+export interface DateRangePickerDropdownProps
+  extends Pick<
+    Required<DateRangePickerBaseProps>,
```

Comment:
Why these props set as required? If some are optional in the original component I'd expect them to remain optional here.

Reply:
In the original dropdown component, it seems we are making some of these optional props required anyway?: https://github.com/cloudscape-design/components/blob/main/src/date-range-picker/dropdown.tsx#L40

It seems many of these optional props are actually required but are defined as optional in the interfaces